app/assets/javascripts/app/orders_edit_date.js

```diff
@@ -0,0 +1,33 @@
+window.addEventListener("DOMContentLoaded", () => {
+  const orderStatusSelect = document.querySelector("#order_status_id");
+
+  if (orderStatusSelect) {
+    toggleFulfilledAt();
+  
+    orderStatusSelect.addEventListener("change", () => toggleFulfilledAt());
+
+    // In the _edit_date partial, show the "Fulfilled at" field only if 
+    // the "Order Status" is "Complete"
+    function toggleFulfilledAt() {
```

Comment:
I got thrown by declaring this after its first use. It's not a big deal, but stylistically, I would recommend moving this outside the event listener so it's declared before its use.